src/types/intersection.ts

```diff
@@ -56,7 +59,7 @@ export class IntersectionType<Types extends OneOrMore<BaseObjectLikeTypeImpl<unk
     readonly combinedName = combinedName(this.types);
     /** {@inheritdoc BaseObjectLikeTypeImpl.possibleDiscriminators} */
     readonly possibleDiscriminators: readonly PossibleDiscriminator[] = this.types
-        .flatMap(t => t.possibleDiscriminators || throwWrongBasicType(this.types))
+        .flatMap(t => t.possibleDiscriminators)
```

Comment:
typescript-eslint complained that we didn't trust the types, which we didn't. What is the point of TypeScript if you don't trust the types? So, removed this.